### PR TITLE
[IMP] mail: silent mail during tests

### PR DESCRIPTION
**Pull request by Xavier-Do**
The mail loggers can be noisy during tests, lets mute them in this scope
While on it, also adapt in_test_mode by a more robust current_test check

### CODE REVIEW

**Comment by Xavier-Do** [comment]
@tde-banana-odoo not final but this mail noise during test is boring, what do you think? Just need to check for the test mode detection changes (mainly how to re-enable the mail server without other side effect).

**Comment by tde-banana-odoo** [comment]
@Xavier-Do Fine for me obviously :) I already have a lot of mute loggers so it annoys me as well :)

File: addons/mail/models/mail_mail.py [hunk 1]
```diff
@@ -571,9 +571,10 @@ def send(self, auto_commit=False, raise_exception=False):
                     smtp_session=smtp_session,
                     alias_domain_id=alias_domain_id,
                 )
-                _logger.info(
-                    'Sent batch %s emails via mail server ID #%s',
-                    len(batch_ids), mail_server_id)
+                if modules.module.current_test:
```
[the remark this answers is not in the document]
Comment (Xavier-Do) [reply addons/mail/models/mail_mail.py]: Yes, thanks!

**Comment by kmagusiak** [comment]
I have a general question about `odoo.modules.module.current_test`. I knew about a similar `getattr(threading.current_thread(), 'testing', False)` and `Registry.in_test_mode()`

What are the differences? If there aren't any, what is the favored way?

**Comment by tde-banana-odoo** [comment]
@Xavier-Do if you want to merge in mail, you have to come back to the team. That's the law.

**Comment by Xavier-Do** [comment]
> I have a general question about odoo.modules.module.current_test. I knew about a similar getattr(threading.current_thread(), 'testing', False) and Registry.in_test_mode()

> What are the differences? If there aren't any, what is the favored way?

Historicaly, there was no good way to detect if we are in test mode, I invite you to check #156852 for more infos.

So since this pr, odoo.modules.module.current_test is the way to go. Before that, `threading.current_thread().testing` was working both in at_install and post_install, but not inside a request.

**Comment by Xavier-Do** [comment]
@robodoo r+

**Pull request closed without merging**